ext4: update ctime and mtime for truncate with extents.

From: Solofo Ramangalahy <Solofo.Ramangalahy@bull.net>

The recently announced "Linux POSIX file system test suite"
catched a truncate issue when using extents:
mtime and ctime are not updated when truncate is successful.

This is the single issue catched with "default" ext4 (mkfs and mount
with minimal options).
The testsuite does not report failure with -o noextents.

With the following patch, all tests of the testsuite passes.

Signed-off-by: Solofo Ramangalahy <Solofo.Ramangalahy@bull.net>
Signed-off-by: Mingming Cao <cmm@us.ibm.com> 

---
 fs/ext4/extents.c |    1 +
 1 file changed, 1 insertion(+)

Index: linux-2.6.25-rc8/fs/ext4/extents.c
===================================================================
--- linux-2.6.25-rc8.orig/fs/ext4/extents.c	2008-04-09 21:01:38.000000000 -0700
+++ linux-2.6.25-rc8/fs/ext4/extents.c	2008-04-09 21:12:19.000000000 -0700
@@ -2808,6 +2808,7 @@ out_stop:
 		ext4_orphan_del(handle, inode);
 
 	up_write(&EXT4_I(inode)->i_data_sem);
+	inode->i_mtime = inode->i_ctime = ext4_current_time(inode);
 	ext4_journal_stop(handle);
 }
 
